### PR TITLE
simplify registry functions

### PR DESCRIPTION
**Pull request by janishorsts**

### CODE REVIEW

File: template/example_test.go [hunk 1]
```diff
@@ -52,10 +52,8 @@ func ExampleTemplate_complexMessage() {
 
 	// Define new function color
 	colorF := registry.Func{
-		Name: "color",
 		FormatSignature: &registry.Signature{
 			// Mark that input/operand is required for a function
```
Comment (mvilks) [inline template/example_test.go:56]: also remove the comment

File: template/registry/datetime.go [hunk 2]
```diff
@@ -11,129 +11,185 @@ import (
...
+
+func parseDatetimeOptions(options Opts) (*datetimeOptions, error) {
+	var (
+		opts datetimeOptions
+		err  error
+	)
+
+	// The predefined date formatting style to use.
+	dateStyles := []string{"full", "long", "medium", "short"}
+	if opts.DateStyle, err = options.GetString("dateStyle", "", dateStyles); err != nil {
```
Comment (mvilks) [inline template/registry/datetime.go:59]: No sensible fallback values?

File: template/registry/number_test.go [hunk 3]
```diff
@@ -11,66 +11,66 @@ func Test_Number(t *testing.T) {
...
 
-	assert = assertFmt(t, numberRegistryFunc, map[string]any{"signDisplay": "never"}, language.AmericanEnglish)
+	assert = assertFormat(t, numberRegistryFunc, map[string]any{"signDisplay": "never"}, language.AmericanEnglish)
 	assert(-0.15, "0.15")
 	assert(0, "0")
 	assert(0.15, "0.15")
 
-	assert = assertFmt(t, numberRegistryFunc, map[string]any{"maximumFractionDigits": 1}, language.AmericanEnglish)
+	assert = assertFormat(t, numberRegistryFunc, map[string]any{"maximumFractionDigits": 1}, language.AmericanEnglish)
 	assert(0.15, "0.1")
```
Comment (mvilks) [inline template/registry/number_test.go:40]: No rounding up?

Comment (janishorsts) [reply template/registry/number_test.go:40]: No. MF2 does not describe the rounding strategy. However, the `number` function borrows ideas from EcmaScript - [Intl.NumberFormat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat).

In Javascript, it does round up by default:

```javascript
console.log(
  new Intl.NumberFormat('en', { maximumFractionDigits: 2 }).format(
    6.785,
  ),
)
// prints "6.79"
```

Comment (janishorsts) [reply template/registry/number_test.go:40]: Golang does not round up by default.

Comment (janishorsts) [reply template/registry/number_test.go:40]: Leaving it as is, the MF2 WG does not define the implementation/formatting details. The options are hints for the implementation.

[More](https://github.com/unicode-org/message-format-wg/blob/1dc84e648a6f98d74ac62306abaacc0bed8e4fc5/meetings/2024/notes-2024-02-14.md)

File: template/registry/datetime.go (hunk 2 above)
Comment (janishorsts) [reply template/registry/datetime.go:59]: https://github.com/unicode-org/message-format-wg/blob/main/spec/registry.xml#L31 no default value defined in MF2 WG.

**Review by mvilks — APPROVED** [review]

**Comment by janishorsts** [comment]
These are transitional changes. More changes are needed over time.

**Pull request merged**